expressEngine/test/customer/post/customer.js

```diff
@@ -0,0 +1,180 @@
+  const { email, contactNo, name, city, country } = success;
+
+  describe(`post function in ${customer}`, () => {
+    it('should register a customer using given email, contact number, name, city and country', async () => {
+      const result = await chai
+        .request(server)
+        .post(customer)
+        .set(header)
+        .send({ email, name, city, country, contactNo });
+      console.log('------------', result.body);
+      result.should.have.status(ok);
+      result.body.should.have.property('data');
+      result.body.should.have.property('message');
+    });
+
+    describe('check for wrong email format', () => {
+      const {
+        empty,
+        withoutAtOperator,
+        withoutAlphabets,
+        withoutPeriod,
+        withoutNumber,
+      } = error.email;
+
+      it('should fail without email', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ name, city, country, contactNo });
+        console.log('----++++++++', result.body);
+        result.should.have.status(internalServerError);
+        result.body.should.have.property('error');
+      });
+
+      it('should fail for blank email', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ email: empty, name, city, country, contactNo });
+        result.should.have.status(internalServerError);
+        result.body.should.have.property('error');
+      });
+
+      it('should fail for email without @ operator', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ email: withoutAtOperator, name, city, country, contactNo });
+        result.should.have.status(internalServerError);
+        result.body.should.have.property('error');
+      });
+
+      it('should fail for email without period(dot) ', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ email: withoutPeriod, name, city, country, contactNo });
+        result.should.have.status(internalServerError);
+        result.body.should.have.property('error');
+      });
+
+      it('should fail for email without alphabetic characters', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ email: withoutAlphabets, name, city, country, contactNo });
+        result.should.have.status(internalServerError);
+        result.body.should.have.property('error');
+      });
+
+      it('should fail for email without number', async () => {
+        const result = await chai
+          .request(server)
+          .post(customer)
+          .set(header)
+          .send({ email: withoutNumber, name, city, country, contactNo });
```

Comment:
@pvvarun error failing and result.body have the message - 'email is required field' and withoutNumber variable is undefined, so test case is not failing because of the number validation for sure.